File: leet_2287_rearrange.cc
class Solution {
public:
    int rearrangeCharacters(string s, string target)
    {
        // way 1 : map
        // unordered_map<char, int> S, T;

        // way 2 : vi of 26
        vector<int> S(26), T(46);
        int res = INT_MAX;
        for (char c : s)
            S[c - 'a']++;
        for (char c : target)
            T[c - 'a']++;
        int i = -1;
        while (++i < 26)
        {
            int Tcount = T[i];
            int Scount = S[i];
            if (T[i])
                res = min(res, Scount / Tcount);
        }
        return res;
    }
};

File: leet_2287_rearrange.go
func rearrangeCharacters(s string, target string) int {

    var T, S [26]int
    i := 0
    for i < 26 {
        T[i], S[i] = 0, 0
        i++
    }
    res := math.MaxInt32
    for _, c := range s {
        S[c - 'a']++
    }
    for _, c := range target {
        T[c - 'a']++
    }

    i = 0
    for i < 26 {
        if T[i] > 0 {
            Scount := S[i]
            Tcount := T[i]
            res = min(res, Scount / Tcount)
        }
        i++
    }
    return res
}
